simpleclient/src/main/java/io/prometheus/client/Histogram.java

```diff
@@ -0,0 +1,262 @@
+package io.prometheus.client;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Histogram metric, to track distributions of events.
+ * <p>
+ * Example of uses for Histograms include:
+ * <ul>
+ *  <li>Response latency</li>
+ *  <li>Request size</li>
+ * </ul>
+ * <p>
+ * <em>Note:</em> Each bucket is one timeseries. Many buckets and/or many dimensions with labels
+ * can produce large amount of time series, that may cause performance problems.
+ * 
+ * <p>
+ * The default buckets are intended to cover a typical web/rpc request from milliseconds to seconds.
+ * <p>
+ * Example Histograms:
+ * <pre>
+ * {@code
+ *   class YourClass {
+ *     static final Histogram requestLatency = Histogram.build()
+ *         .name("requests_latency_seconds").help("Request latency in seconds.").register();
+ *
+ *     void processRequest(Request req) {  
+ *        Histogram.Timer requestTimer = requestLatency.startTimer();
+ *        try {
+ *          // Your code here.
+ *        } finally {
+ *          requestTimer.observeDuration();
+ *        }
+ *     }
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * You can choose your own buckets:
+ * <pre>
+ * {@code
+ *     static final Histogram requestLatency = Histogram.build()
+ *         .buckets(.01, .02, .03, .04)
+ *         .name("requests_latency_seconds").help("Request latency in seconds.").register();
+ * }
+ * </pre>
+ * {@link Histogram.Builder#linearBuckets(double, double, int) linearBuckets} and
+ * {@link Histogram.Builder#exponentialBuckets(double, double, int) exponentialBuckets}
+ * offer easy ways to set common bucket patterns.
+ */
+public class Histogram extends SimpleCollector<Histogram.Child, Histogram> {
+  double[] buckets;
+
+  Histogram(Builder b) {
+    super(b);
+    buckets = b.buckets;
+    initializeNoLabelsChild();
+  }
+
+  public static class Builder extends SimpleCollector.Builder<Builder> {
+    double[] buckets = new double[]{.005, .01, .025, .05, .075, .1, .25, .5, .75, 1, 2.5, 5, 7.5, 10};
```

Comment:
I was trying to get reasonable granularity in an understandable way that doesn't result in too many buckets.

This will get tuned after I start using it more seriously.